plugins/linuxplugin/nsplugin/microservices.go

```diff
@@ -103,24 +103,31 @@ func (plugin *NsHandler) HandleMicroservices(ctx *MicroserviceCtx) {
 	}
 	ctx.created = nextCreated
 
-	// Finally inspect newly created containers.
+	// Inspect newly created containers
 	listOpts := docker.ListContainersOptions{
 		All:     true,
 		Filters: map[string][]string{},
 	}
+	// List containers and filter all older than 'since' ID
 	if ctx.since != "" {
 		listOpts.Filters["since"] = []string{ctx.since}
 	}
-
 	containers, err = plugin.dockerClient.ListContainers(listOpts)
 	if err != nil {
-		plugin.Log.Errorf("Error listing docker containers: %v", err)
-		if err, ok := err.(*docker.Error); ok &&
-			(err.Status == 500 || err.Status == 404) { // 404 is required to support older docker version
-			plugin.Log.Debugf("clearing since: %v", ctx.since)
-			ctx.since = ""
+		// If 'since' container was not found, list all containers (404 is required to support older docker version)
+		if dockerErr, ok := err.(*docker.Error); ok && (dockerErr.Status == 500 || dockerErr.Status == 404) {
+			// Reset filter and list containers again
+			plugin.Log.Debug("clearing 'since' %s", ctx.since)
+			listOpts.Filters = map[string][]string{}
```

Comment:
It would be better to just remove `"since"` from the map like this: `delete(listOpts.Filters, "since")`. This way in the future if some field is added it's not removed by this.